modules/caddytls/connpolicy.go

```diff
@@ -461,6 +470,31 @@ func setDefaultTLSParams(cfg *tls.Config) {
 	cfg.PreferServerCipherSuites = true
 }
 
+// LeafVerificationValidator implements custom client certificate verification.
+// It is intended for installation only by clientauth.ConfigureTLSConfig().
+type LeafVerificationValidator struct {
```

Comment:
```suggestion
type LeafCertValidator struct {
```

or maybe better:

```suggestion
type LeafCertClientAuth struct {
```

But let's see what we decide in #4518 first